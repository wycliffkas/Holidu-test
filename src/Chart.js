import React from "react";
import {
  Bar,
  BarChart,
  Label,
  ResponsiveContainer,
  XAxis,
  YAxis,
} from "recharts";
import Title from "./Title";

export default function Chart({ data, countryDisplay }) {
  return (
    <React.Fragment>
      <Title>Score statistics</Title>
      <ResponsiveContainer>
        <BarChart
          data={data}
          margin={{
            top: 16,
            right: 16,
            bottom: 20,
            left: 24,
          }}
        >
          <XAxis dataKey={countryDisplay ? "country" : "gender"}>
            <Label
              dy={25}
              angle={0}
              position="center"
              style={{ textAnchor: "left" }}
            >
              {countryDisplay ? "Country" : "Gender"}
            </Label>
          </XAxis>
          <YAxis>
            <Label angle={270} position="left" style={{ textAnchor: "middle" }}>
              Average score
            </Label>
          </YAxis>
          <Bar type="monotone" dataKey="score" fill="#556CD6" />
        </BarChart>
      </ResponsiveContainer>
    </React.Fragment>
  );
}
