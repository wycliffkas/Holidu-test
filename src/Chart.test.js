import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Chart from "./Chart";

configure({ adapter: new Adapter() });

describe("<Chart /> ", () => {
  it("chart should render correctly", () => {
    const container = shallow(<Chart />);
    expect(container).toMatchSnapshot();
  });
});
