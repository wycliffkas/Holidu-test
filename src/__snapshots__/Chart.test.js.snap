// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chart />  chart should render correctly 1`] = `ShallowWrapper {}`;
